.github/actions/setup-lxd-cluster/lxd-cluster.sh

```diff
@@ -0,0 +1,372 @@
+  security.nesting: "true"
+devices:
+  kvm:
+    source: /dev/kvm
+    type: unix-char
+  vhost-net:
+    source: /dev/vhost-net
+    type: unix-char
+  vhost-vsock:
+    source: /dev/vhost-vsock
+    type: unix-char
+  vsock:
+    mode: "0666"
+    source: /dev/vsock
+    type: unix-char
+EOF
+    fi
+
+    # Setup cluster instances.
+    for i in $(seq 1 "${CLUSTER_SIZE}"); do
+        instance="${INSTANCE}-${i}"
+
+        local state
+        state=$(lxc list --format csv --columns s "${instance}")
+
+        case "${state}" in
+        "RUNNING")
+            echo "Instance ${instance} already running."
+            continue
+            ;;
+        "STOPPED")
+            echo "Starting instance ${instance}..."
+            lxc start "${instance}"
+            continue
+            ;;
+        esac
+
+        local args=(--profile container-kvm --config "security.nesting=true")
+        local ifName="eth0"
+        if [ "${INSTANCE_TYPE}" = "virtual-machine" ]; then
+            args=(--vm)
+            ifName="enp5s0"
+        fi
+
+        echo "Creating instance ${instance} ..."
+
+        lxc init "${INSTANCE_IMAGE}" "${instance}" \
+            --storage "${STORAGE_POOL}" \
+            --network "${NETWORK_NAME}" \
+            --config limits.cpu=4 \
+            --config limits.memory=4GiB \
+            --config security.devlxd.images="true" \
+            "${args[@]}"
+
+        # Apply network bridge configuration.
+        lxc config set "${instance}" cloud-init.network-config - <<EOF
+version: 2
+ethernets:
+  ${ifName}:
+    dhcp4: false
+bridges:
+  br0:
+    interfaces: [${ifName}]
+    dhcp4: true
+EOF
+
+        lxc start "${instance}"
+
+        if [ "${STORAGE_DRIVER}" = "zfs" ] && [ "${INSTANCE_TYPE}" = "container" ]; then
+            # Attach additional disk to each container to allow creating delegated ZFS storage pool
+            # within the cluster.
+            local disk="${instance}-disk"
+            lxc storage volume create "${STORAGE_POOL}" "${disk}"
+            lxc config device add "${instance}" "${disk}" disk pool="${STORAGE_POOL}" source="${disk}" path=/mnt/disk
+        fi
+    done
+
+    # Wait for instances to become ready.
+    for i in $(seq 1 "${CLUSTER_SIZE}"); do
+        instance="${INSTANCE}-${i}"
+        waitInstanceReady "${instance}"
+        lxc exec "${instance}" -- systemctl is-system-running --wait
+    done
+
+    # Install LXD on VMs.
+    for i in $(seq 1 "${CLUSTER_SIZE}"); do
+        instance="${INSTANCE}-${i}"
+
+        echo "Preparing instance ${instance} ..."
+
+        # Install snap daemon.
+        lxc exec "${instance}" --env=DEBIAN_FRONTEND=noninteractive -- apt-get update
+        lxc exec "${instance}" --env=DEBIAN_FRONTEND=noninteractive -- apt-get -qq -y install snapd
+
+        # Install LXD snap.
+        lxc exec "${instance}" -- snap install lxd --channel "${VERSION_LXD}" || lxc exec "${instance}" -- snap refresh lxd --channel "${VERSION_LXD}"
```

Comment:
We don't care about 22.04 and earlier, right?
```suggestion
        lxc exec "${instance}" -- snap install lxd --channel "${VERSION_LXD}"
```